src/compile/axis/rules.ts

```diff
@@ -136,9 +137,11 @@ export function domainAndTicks(property: keyof VgAxis, specifiedAxis: Axis, isGr
   return specifiedAxis[property];
 }
 
-export function labelOverlap(fieldDef: FieldDef<string>, specifiedAxis: Axis, channel: Channel, isGridAxis: boolean) {
-  // TODO: use true for non-log continuous scales, and use "greedy" for log scales
+export function labelOverlap(fieldDef: FieldDef<string>, specifiedAxis: Axis, channel: Channel, isGridAxis: boolean, scaleType: ScaleType) {
   if (!isGridAxis && channel === 'x' && !labelAngle(specifiedAxis, channel, fieldDef)) {
+    if (scaleType === 'log') {
```

Comment:
@domoritz Actually why is this rule for log only applied for x?  (I guess y's log can similar has overlapping labels?)